src/index.js

```diff
@@ -387,8 +369,17 @@ Penpal.connectToParent = ({ parentOrigin, methods = {} }) => {
       child.removeEventListener(MESSAGE, handleMessageEvent);
       reject('Child: Connection destroyed');
     })
-  });
 
+    const sendHandshakeMessage = function() {
+      parent.postMessage({
+	    penpal: HANDSHAKE,
+	    methodNames: Object.keys(methods),
+      }, document.referrer);
```

Comment:
I think it would be wise to do a check before sending the handshake message and if the origin derived from `document.referrer` doesn't match the `parentOrigin` the consumer passed in as an argument (see https://github.com/Aaronius/penpal/blob/8c8a330c1d7a484d2a5261a6060a2f46f43e6c16/src/index.js#L358-L359 for an example of how this is done) then we throw an error stating such. At the moment, the handshake-reply message would be ignored in this case (because the origins wouldn't match), but I wonder if there's any security implications of even sending the initial handshake message to a nefarious website. In any case, throwing an error before sending the handshake message would probably help consumers debug the handshake failure as well.